package.json

```diff
@@ -35,13 +37,9 @@
     "grunt-contrib-watch": "^1.1.0",
     "grunt-shell": "^3.0.1",
     "grunt-ts": "^6.0.0-beta.22",
-    "typescript": "^3.9.5"
-  },
-  "dependencies": {
     "@popperjs/core": "^2.4.0",
     "@stackoverflow/stacks-icons": "^2.8.0",
-    "backstopjs": "^5.0.1",
-    "eleventy-plugin-nesting-toc": "^1.2.0",
-    "stimulus": "^1.1.1"
+    "stimulus": "^1.1.1",
+    "typescript": "^3.9.5"
```

Comment:
`typescript` needs to be a `devDependency` instead

Reply:
Everything is now a devDependency. The diff looks a little weird but there are no `dependencies` anymore.

Reply:
Oh, yeah.... that's correct. We want to instead move anything that gets included in the final build into `dependencies`. So, popperjs, icons, stimulus, etc all need to be dependencies.

Reply:
Why? Popper, Stimulus, etc. are all compiled into JS, right? Anything consuming Stacks will only be using the compiled JS.

Reply:
As I understand it, you only need those dependencies when developing _for_ Stacks, not consuming it.

Reply:
It's the opposite logic. There's two major ways to consume a package:

1. using the fully compiled script in your browser
  * dependencies don't matter at all because, hey, it's all compiled for you
2. importing into another project for use
  * `devDependencies` says "these are for developing ON Stacks"
  * `dependencies` says "these are needed to USE Stacks"
  * Anything missing from `dependencies` will not get included into the user's project and Stacks will fail to compile

Think of it as: "will Stacks work without this package?" and anything answered **no** will go into `dependencies`.

Reply:
So, I'm actually partially wrong: `stacks-icons` is not actually used in Stacks, but *only* in the docs, so it would go into `devDependencies`. Missed that one when writing them out.

Reply:
I think I'm missing something with our JS side of things. We don't deliver JS source, only compiled `stacks.js` and that's by design. We can't compile JS even if we include Popper or Stimulus, because we don't include JS source files.

https://unpkg.com/browse/@stackoverflow/stacks@0.52.1/

I'm happy with the path forward being "Here's compiled everything, CSS and JS. Also here are Less files if you need to compile the CSS yourself :v:" and have redesigned `package.json` to reflect this.

Reply:
I see your point, but this is the "correct" way of doing things. It gives a clean separation of "what goes into the bundle and what doesn't". Other than my listed reasons, there are many other reasons why this is a good idea, including (but not limited to): 

* security
  * immediately know what we're shipping to consumers (notably, stackoverflow.com!)
* code debt / cleanliness
  * same as security, but so we ship less obsolete bytes
* typings
  * allows consumers to use the typescript types from our library
* future proofing / flexibility
  * include the whole library, but bundle it inside another larger bundle
  * possibly split out common dependencies so you don't ship them twice
  * eventually, someone will want to import single components

I can probably come up with some more if you press me hard enough 😛

Reply:
"This" meaning, including all JS source files and adding `Popper` and `Stimulus` as `Dependencies`?

Reply:
We don't need to include all JS source files at the moment (it would be nice to have them later), but you will need to:

* move both stimulus and popper into dependencies
* run `npm i` to update `package.lock`